src/main/java/ee/carlrobert/codegpt/settings/state/util/CommonSettings.java

```diff
@@ -0,0 +1,31 @@
+package ee.carlrobert.codegpt.settings.state.util;
+
+import com.intellij.util.xmlb.annotations.Transient;
+import ee.carlrobert.codegpt.credentials.CredentialsManager;
+
+/**
+ * Settings that all service settings share in common.
+ */
+public class CommonSettings<T extends CredentialsManager> {
```

Comment:
Can we remove this class? I don't see any reason to store the `credentialsManager` in the settings state and marking the field as transient seems just a hack to prevent the field from serialization.

Reply:
I dont think using `@Transient` is hacky, the annotation is exactly for the purpose of not wanting particular fields to be serialized, which is the case here.

The `credentialsManager` is in the settings State classes because the Settings Forms also return credentials (e.g. apiKey), as you can see here:
https://github.com/carlrobertoh/CodeGPT/blob/d0dc14ef02cda5d16df2186f982b37800de6482a/src/main/java/ee/carlrobert/codegpt/settings/service/AzureServiceForm.java#L138-L151

The forms return ready-only Settings which can then simply be applied, checked for modification and also be reset in:
https://github.com/carlrobertoh/CodeGPT/blob/d0dc14ef02cda5d16df2186f982b37800de6482a/src/main/java/ee/carlrobert/codegpt/settings/state/AzureSettingsState.java#L61-L101

Removing the `credentialsManager` from the states would mean a lot copy-paste. You also have to keep in mind, that for the `LLama` Service we have 2 `CredentialsManager`s, one for the remote and one for the local config.

Reply:
> The credentialsManager is in the settings State classes because the Settings Forms also return credentials (e.g. apiKey), as you can see here:

I understand, but why do we want to persist `credentialManager` in the state in the first place? I might be misunderstanding something here, but what value does it actually bring?

The [separate state class](https://plugins.jetbrains.com/docs/intellij/persisting-state-of-components.html#d264d665_40) approach seems to decouple the state and the settings service logic quite nicely. If we just need to access the credentials manager from each of these settings, wouldn't it be simpler to just define a single getter in the service logic?

For example:
```java
public AzureCredentialsManager getCredentialsManager() {
  return ApplicationManager.getApplication().getService(AzureCredentialsManager.class);
}
```

Reply:
I separated the `CredentialsManager`s from the State's, and converted every `CredentialsManager` into a Service with a `getInstance` method, for Llama I added `LlamaLocalCredentialsManager` + `LlamaRemoteCredentialsManager`. Looks cleaner now